src/observables/OptionStream.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Options } from "../omnisharp/options";
+import { Subject } from "rxjs/Subject";
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { vscode } from "../vscodeAdapter";
+import 'rxjs/add/operator/take';
+import Disposable from "../Disposable";
+
+export default class OptionStream {
+    private optionStream: BehaviorSubject<Options>;
+    private optionChangeDebouncer: Subject<void>;
+
+    constructor(vscode: vscode) {
+        this.optionStream = new BehaviorSubject<Options>(Options.Read(vscode));
+        this.optionChangeDebouncer = new Subject<void>();
+        this.optionChangeDebouncer.debounceTime(1000).subscribe(() => { this.optionStream.next(Options.Read(vscode)); });
```

Comment:
Is the subscription to the options change event coming in a later PR? Might make sense to include that in this PR so we can see that whole flow.

Reply:
Oh, I see you did it below.
If this class takes a vscode in its constructor, it should handle all the event subscribing itself.

Reply:
We are adding the subscription disposable to the set of disposables of the extension so that the resources are freed up when the vscode window closes. Hence I think it will be better to keep the subscription to the `vscode.onDidChangeConfiguration` inside the main.ts only where other similar subscriptions are present. Thoughts ?